libs/collectives/CMakeLists.txt

```diff
@@ -11,27 +11,59 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 set(collectives_headers
   hpx/collectives/all_reduce.hpp
   hpx/collectives/all_to_all.hpp
+  hpx/collectives/barrier.hpp
+  hpx/collectives/broadcast.hpp
+  hpx/collectives/fold.hpp
+  hpx/collectives/gather.hpp
+  hpx/collectives/latch.hpp
+  hpx/collectives/reduce.hpp
+  hpx/collectives/detail/barrier_node.hpp
+  hpx/collectives/detail/latch.hpp
+)
+
+# Default location is $HPX_ROOT/libs/collectives/include_compatibility
+set(collectives_compat_headers
+  hpx/lcos/barrier.hpp
+  hpx/lcos/broadcast.hpp
+  hpx/lcos/fold.hpp
+  hpx/lcos/gather.hpp
+  hpx/lcos/latch.hpp
+  hpx/lcos/reduce.hpp
 )
 
 # Default location is $HPX_ROOT/libs/collectives/src
 set(collectives_sources
+  barrier.cpp
+  latch.cpp
+  detail/barrier_node.cpp
 )
 
 include(HPX_AddModule)
 add_hpx_module(collectives
+    COMPATIBILITY_HEADERS ON    # Added in 1.4.0
     DEPRECATION_WARNINGS
     FORCE_LINKING_GEN
     SOURCES ${collectives_sources}
     HEADERS ${collectives_headers}
+    COMPAT_HEADERS ${collectives_compat_headers}
+    EXCLUDE_FROM_GLOBAL_HEADER
+      hpx/collectives/detail/barrier_node.hpp
+      hpx/collectives/detail/latch.hpp
     DEPENDENCIES
+      hpx_allocator_support
       hpx_assertion
+      hpx_cache
       hpx_concurrency
       hpx_config
       hpx_errors
       hpx_format
+      hpx_iterator_support
       hpx_logging
+      hpx_parallel_executors
       hpx_preprocessor
+      hpx_program_options
```

Comment:
Does the module itself need `program_options` or just the tests?

Reply:
The module depends on things in the big runtime-blob which in turn depend on program_options. I had to add all those dependencies here in order to be able to compile things. I did not try to disentangle this at this point.